app/src/main/java/org/mozilla/focus/whatsnew/WhatsNew.kt

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.whatsnew
+
+import android.content.Context
+import android.preference.PreferenceManager
+import android.support.annotation.VisibleForTesting
+import org.mozilla.focus.ext.appVersionName
+
+class WhatsNew {
+    companion object {
+        // TODO: Update to final URL
+        const val SUMO_URL = "https://support.mozilla.org/en-US/products/focus-firefox/firefox-focus-android"
+
+        @VisibleForTesting const val PREFERENCE_KEY_APP_NAME = "lastKnownAppVersionName"
+        private const val PREFERENCE_KEY_SESSION_COUNTER = "updateSessionCounter"
+        private const val SESSIONS_PER_UPDATE = 3
+
+        @VisibleForTesting var wasUpdatedRecently: Boolean? = null
+
+        fun wasUpdatedRecently(context: Context): Boolean {
+            if (wasUpdatedRecently == null) {
+                wasUpdatedRecently = when {
+                    hasBeenUpdated(context) -> {
+                        rememberAppVersionName(context)
+                        setSessionCounter(context)
+                        true
+                    }
+                    else -> hasBeenUpdatedInRecentSession(context)
+                }
+            }
+            return wasUpdatedRecently!!
+        }
+
+        @JvmStatic
+        fun reset(context: Context) {
+            wasUpdatedRecently = false
+            setSessionCounter(context, 0)
+        }
+
+        private fun hasBeenUpdated(context: Context): Boolean {
+            val lastKnownAppVersionName = getLastKnownAppVersionName(context)
+            if (lastKnownAppVersionName.isNullOrEmpty()) {
+                rememberAppVersionName(context)
+                return false
+            }
+
+            return !lastKnownAppVersionName.equals(context.appVersionName)
+        }
+
+        private fun getLastKnownAppVersionName(context: Context): String? {
+            return PreferenceManager.getDefaultSharedPreferences(context)
+                    .getString(PREFERENCE_KEY_APP_NAME, null)
+        }
+
+        private fun rememberAppVersionName(context: Context) {
```

Comment:
nit: `remember` -> `set`, `save`, or `store`? It'd be more conventional

Also, because of `getLastKnownAppVersionName`, I think this should be `...LastKnownAppVersionName` too.